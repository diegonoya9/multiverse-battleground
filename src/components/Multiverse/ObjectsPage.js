import classes from "./ObjectsPage.module.css"
import { memo, useEffect, useState } from "react";
import ReactAudioPlayer from 'react-audio-player';
import musicFile from "../../assets/sounds/music/DiscoEterno.WAV"
const ObjectsPage = ({ user, changeMultiverseActivePage }) => {
    const audioStyle = {
        display: 'none',
    };
    return (<div>
        <ReactAudioPlayer src={musicFile} /*autoPlay*/ controls style={audioStyle} />
        <button value="Back to Main Menu" className={classes.backToMainMenuBtn} onClick={() => { changeMultiverseActivePage("mainMenu") }} >Back to Main Menu </button>
        <div className={classes.container} >
            {user &&
                user.objects.map((object) => {
                    return (
                        <div className={classes.objectContainer} key={object.name}>
                            <span className={classes.objectName}>{object.name}</span>
                            <div className={classes.imageContainer}>
                                <img alt="fighter" src={object.img} className={classes.objectImg} />
                            </div>
                            <span className={classes.objectName}>Quantity:{object.quantity}</span>
                        </div>
                    );
                })}
        </div>
    </div>
    );

}

export default memo(ObjectsPage)