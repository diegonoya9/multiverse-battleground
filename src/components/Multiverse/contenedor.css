* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: sans-serif;
}

.body {
    background: #282c34;
}

.container {
    margin-top: 2em;
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    min-height: 100vh;
    background: #282c34;
    gap: 3em;
}

.card {
    position: relative;
    width: 300px;
    height: 350px;
}

.card .face {
    position: absolute;
    width: 100%;
    height: 100%;
    backface-visibility: hidden;
    border-radius: 10px;
    overflow: hidden;
    transition: .5s;
}

.card .front {
    transform: perspective(600px) rotateY(0deg);
    box-shadow: 0 5px 10px #000;
}

.title {
    text-transform: capitalize;
}

.card .front img {
    position: absolute;
    width: 40%;
    height: 50%;
    object-fit: cover;
}

.card .front h3 {
    position: absolute;
    bottom: 0;
    width: 100%;
    padding: 10px;
    height: 45px;
    color: #fff;
    background: rgba(0, 0, 0, .4);
    text-align: center;
}

.card .back {
    transform: perspective(600px) rotateY(180deg);
    background-color: #1b1d20;
    padding: 15px;
    color: #fff;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    text-align: center;
    box-shadow: 0 5px 10px #000;
}

.card .back button {
    border-top: solid 1px #f3f3f3;
    height: 25px;
    line-height: 25px;
}

.card .back p {
    font-size: 14px;
    margin-top: 5px;
    letter-spacing: 2px;
}

.card:hover .front {
    transform: perspective(600px) rotateY(180deg);
}

.card:hover .back {
    transform: perspective(600px) rotateY(360deg);
}