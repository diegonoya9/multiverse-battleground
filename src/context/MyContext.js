import React, { createContext, useState } from 'react';

const MyContext = createContext();

const MyContextProvider = ({ children }) => {
    const [userContext, setUserContext] = useState({
        idUsuario: 1,
        userName: "",
        backEndUrl: process.env.NODE_ENV === 'development' ? "http://localhost:3009/api" : "https://multiverse-battlegorund-back.onrender.com/api" // Otros datos relacionados con el usuario si es necesario
    });

    const setUserId = (newId) => {
        setUserContext({
            ...userContext,
            idUsuario: newId,
        });
    };
    const setUserName = (userName) => {
        setUserContext({
            ...userContext,
            userName
        });
    };

    return (
        <MyContext.Provider value={{ userContext, setUserId, setUserName }}>
            {children}
        </MyContext.Provider>
    );
};

export { MyContext, MyContextProvider };
