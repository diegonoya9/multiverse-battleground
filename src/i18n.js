import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

i18n
  // detect user language
  // learn more: https://github.com/i18next/i18next-browser-languageDetector
  .use(LanguageDetector)
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  // for all options read: https://www.i18next.com/overview/configuration-options
  .init({
    debug: false,
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, // not needed for react as it escapes by default
    },
    resources: {
      en: {
        translation: {
          multiverse: {
            fight: 'Fight',
            welcome: 'Welcome',
            money: 'You have',
            bag: "Bag",
            shop: "Shop",
            users: "Users",
            fighters: "Fighters",
            config:"Settings"
          },
          fighterspage: {
            removeFromParty: "Remove from party",
            addToParty: "Add to party",
            setFirst: "Set First",
            back: "Back to Main Menu",
            viewMovements: "View Movements"
          },
          home:{
            welcome:"Welcomeas to the Multiverse Battleground",
            getReady:"Get ready for epic battles!",
            start:"Start Adventure!"
          },
          objectspage:{
            quantity:"Quantity",
            description:"Description"
          },
          settingspage:{
            back:"Back to Main Menu",
            bg:"Background Music volume",
            sound:"Sounds volume",
            sfx:"SFX volume",
            save:"Save changes"
          }
        }
      },
      es: {
        translation: {
          multiverse: {
            fight: 'A pelear',
            welcome: 'Hola',
            money: 'Tienes',
            bag: "Mochila",
            shop: "Tienda",
            users: "Usuarios",
            fighters: "Luchadores",
            config:"Opciones"
          },
          fighterspage: {
            removeFromParty: "Quitar del equipo",
            addToParty: "Agregar al equipo",
            setFirst: "Seleccionar primero",
            back: "Volver al Menú Principal",
            viewMovements: "Ver Ataques"
          },
          home:{
            welcome:"Bienvenido al Multiverse Battleground",
            getReady:"Preparate para batallas épicas!!",
            start:"Empezar Aventura!"
          },
          objectspage:{
            quantity:"Cantidad",
            description:"Descripción"
          },
          settingspage:{
            back:"Volver al Menú Principal",
            bg:"Volúmen de Música de ambiente",
            sound:"Volúmen de sonidos de interfaz",
            sfx:"Volúmen de Efectos Especiales",
            save:"Guardar cambios"
          }
        }
      },
    }
  });
/*i18n.changeLanguage('es')
  .then(() => {
    console.log('Idioma cambiado a inglés');
    // Puedes realizar acciones adicionales después de cambiar el idioma si es necesario
  })
  .catch(error => {
    console.error('Error al cambiar el idioma', error);
  });*/
export default i18n;
