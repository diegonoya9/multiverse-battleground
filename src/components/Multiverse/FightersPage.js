import classes from "./FightersPage.module.css"
import ReactAudioPlayer from 'react-audio-player';
import Modal from "../UI/Modal";
import { useState, useContext, useEffect, useCallback } from "react";
import musicFile from "../../assets/sounds/music/DirtyLove.WAV"
import Button from "../UI/Button";
import { MyContext } from "../../context/MyContext";
import FighterCard from "../UI/FighterCard";
import { useTranslation } from 'react-i18next';
const FightersPage = ({ user, changeMultiverseActivePage }) => {
    const { t } = useTranslation();
    const priceTags = document.querySelectorAll(".fighterPriceBlock")
    priceTags.forEach(tag => {
        tag.style.display = "none"
    });;
    const [showModal, setShowModal] = useState(false)
    const [modalContent, setModalContent] = useState()
    const { userContext } = useContext(MyContext);
    const [fighters, setFighters] = useState()
    const [moves, setMoves] = useState()
    /*const [showConfirm, setShowConfirm] = useState(false)
    const [userFighterId, setUserFighterId] = useState(false)*/
    const [allowCloseModal, setAllowCloseModal] = useState(true)
    let backEndUrl = userContext.backEndUrl
    let activeUser = userContext.idUsuario
    const audioStyle = {
        display: 'none',
    };
    const closeModal = () => {
        if (allowCloseModal) {
            setShowModal(false)
            setModalContent()
        }
    }
    const addToParty = (userFighterId) => {
        let cant = 0
        let cantExceeded = false
        /* Checks if there are already 4 fighters in the party */
        fighters.forEach((fighter) => {
            if (fighter.in_party === "true") {
                cant++
            }
            if (cant === 4) {
                cantExceeded = true
            }
        })
        if (cantExceeded) {
            setAllowCloseModal(true)
            setModalContent("Can't add more than 4 members to the party.")
            setShowModal(true)
        } else {
            const parameters = [{
                user_fighter_id: userFighterId
            }]
            setModalContent('Adding to party...')
            setAllowCloseModal(false)
            setShowModal(true)
            fetch(backEndUrl + "/addtoparty", {
                method: 'POST', // O 'PUT' si deseas sobrescribir completamente los datos del usuario
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify(parameters),
            }).then(() => {
                updateFighters()
            })
        }
    }
    const removeFromParty = (user_fighter_id) => {
        let newUser = user
        const parameters = [{
            user_fighter_id,
            user_id: newUser.user_id
        }]
        setModalContent('Removing from party...')
        setAllowCloseModal(false)
        setShowModal(true)
        fetch(backEndUrl + "/removefromparty", {
            method: 'POST', // O 'PUT' si deseas sobrescribir completamente los datos del usuario
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(parameters),
        }).then(() => updateFighters())

    }
    const viewActions = (move_id) => {
        let actions = []
        moves.forEach((move) => {
            if (move.move_id === move_id) {
                actions.push(move.actionmoves)
            }
        })
        setAllowCloseModal(true)
        if (actions[0]) {
            setModalContent(actions[0].map((action, index) => {
                return <div key={Math.random()} >
                    <p>Action: {index + 1}</p>
                    <p>Inflicted on: {action.inflicted_on}</p>
                    <p>Type: {action.attack_type}</p>
                    <p>Field: {action.field}</p>
                    <p>Value: {action.value}</p>
                </div>
            }))
            setShowModal(true)
        }
    }
    useEffect(() => {
        if (moves) {
            setModalContent(
                <ul>{moves.map((move) => {
                    return <Button key={move.name} onClick={() => { viewActions(move.move_id) }}>
                        {move.name}
                    </Button>
                })} </ul>
            )
        }
    }, [moves])
    const viewMovements = (user_fighter_id) => {
        let moves = []
        fighters.forEach((fighter) => {
            if (fighter.user_fighter_id === user_fighter_id) {
                fighter.moves.forEach((move) => {
                    moves.push(move.moves)
                })
            }
        })
        setAllowCloseModal(true)
        setMoves(moves)
    }
    useEffect(() => {
        if (modalContent) {
            setShowModal(true)
        }
    }, [modalContent])
    const setFirstFighter = (user_fighter_id) => {
        const parameters = [{
            user_id: user.user_id,
            user_fighter_id
        }]
        setModalContent('Assigning first fighter')
        setShowModal(true)
        setAllowCloseModal(false)
        fetch(backEndUrl + "/setfirstfighter", {
            method: 'POST', // O 'PUT' si deseas sobrescribir completamente los datos del usuario
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(parameters),
        }).then(() => {
            updateFighters()
        })
    }
    const deleteFighter = (userFighterId) => {
       /* setUserFighterId(userFighterId)
        setShowConfirm(true)
        setShowModal(true)*/
    }
    const deleteUserFighter = async (userFighterId) => {
        /* let deleteFighter = await user.fighters.filter((fighter) => {
             return fighter.userFighterId === userFighterId
         })
         let newFighters = user.fighters.filter((fighter) => {
             return fighter.userFighterId !== userFighterId
         })
         const sell = user.objects[4].quantity
         let price = deleteFighter[0].price
         const result = sell + price
         let newUser = user
         newUser.objects[4].quantity = result
         newUser.fighters = newFighters
         setShowModal(false)
         setShowConfirm(false)*/
    }
    const fetchData = async () => {
        try {
            if (backEndUrl && activeUser) {
                const response = await fetch(backEndUrl + '/alluserfighters/' + activeUser);
                const data = await response.json();
                setFighters(data);
            }
        } catch (error) {
            console.error('Error fetching data:', error);
        }
    };
    const updateFighters = async () => {
        await fetchData()
    }

    useEffect(() => {
        setShowModal(false)
        setModalContent()
    }, [fighters])
    useEffect(() => {
        updateFighters()
    }, [])
    return (<div alt="divContainerFightersPage" className={`${classes.body} ${classes.backgroundImg}`}>
        <Button colorType="lightgreen" value={t('fighterspage.back')} onClick={() => { changeMultiverseActivePage("mainMenu") }}></Button>
        <div className={classes.container} >
            <ReactAudioPlayer src={musicFile} autoPlay controls style={audioStyle} />
            {fighters &&
                fighters.map((fighter, i) => {
                    return (<div className={classes.fighterContainer} key={fighter.user_fighter_id}>
                        <FighterCard fighter={fighter}></FighterCard>
                        <Button onClick={() => { if (fighter.in_party === "true") { setFirstFighter(fighter.user_fighter_id) } else { setAllowCloseModal(true); setModalContent('You need to add the fighter to the party first.') } }} value={t('fighterspage.setFirst')}></Button>
                        <Button onClick={() => { viewMovements(fighter.user_fighter_id) }} value={t('fighterspage.viewMovements')}></Button>
                        {fighter.in_party === "true" ?
                            <Button onClick={() => { removeFromParty(fighter.user_fighter_id) }} value={t('fighterspage.removeFromParty')}></Button>
                            :
                            <Button onClick={() => { addToParty(fighter.user_fighter_id) }} value={t('fighterspage.addToParty')}></Button>
                        }
                        {/* <Button onClick={() => { deleteFighter(fighter.user_fighter_id) }} value="Sell Fighter"></Button> */}
                    </div>
                    );
                })}
        </div>
        {/*showModal && !modalContent && <Modal styleType={"battlegroundColiseum"} onClose={closeModal} color="white">
            {showConfirm && <div>
                <h3>Are you sure you want to sell this fighter?</h3>
                <Button onClick={() => deleteUserFighter(userFighterId)}>Sell</Button>
            </div>}
            </Modal>*/}
        {showModal && modalContent && <Modal styleType={"battlegroundColiseum"} onClose={closeModal} color="white">
            {modalContent}
        </Modal>}
    </div >
    );

}

export default FightersPage