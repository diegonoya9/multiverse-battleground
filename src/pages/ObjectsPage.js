import classes from "./ObjectsPage.module.css"
import { memo, useEffect, useState, useContext } from "react";
import Button from "../components/UI/Button";
import ObjectCard from "../components/UI/ObjectCard";
import { MyContext } from "../context/MyContext";
import { useNavigate } from "react-router-dom";
const ObjectsPage = () => {
    const navigate = useNavigate()
    const [userObjects, setUserObjects] = useState()
    const { userContext } = useContext(MyContext);
    let backEndUrl = userContext.backEndUrl
    let user = userContext.user
    useEffect(() => {
        const fetchData = async () => {
            try {
                if (backEndUrl && user) {
                    const response = await fetch(backEndUrl + '/alluserobjects/' + user.user_id);
                    const data = await response.json();
                    setUserObjects(data);
                }
            } catch (error) {
                console.error('Error fetching data:', error);
            }
        };

        fetchData();
    }, [backEndUrl, user])
    return (<div className={classes.backgroundImg}>
        <Button colorType="lightgreen" value="Back to Main Menu" onClick={() => { navigate('/') }}></Button>
        <div className={classes.container} >
            {userObjects &&
                userObjects.map((object) => {
                    return (
                        object.objects.name !== "Money" &&
                        <div className={classes.objectContainer} key={object.name}>
                            <ObjectCard showPrice={false} object={object}></ObjectCard>
                        </div>
                    );
                })}
        </div>
    </div>
    );

}

export default memo(ObjectsPage)