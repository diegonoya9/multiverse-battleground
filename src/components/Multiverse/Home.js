import { useState } from "react"
import Multiverse from "./Multiverse"
import FightModal from "./FightModal"
import Battleground from "../Fights/Battleground"
import classes from './Home.module.css'

const Home = () => {
    const [showModal, setShowModal] = useState(true);

    const handleStartAdventure = () => {
        console.log(showModal)
        // Aquí puedes añadir lógica para iniciar la aventura
        console.log('Adventure started!');
        setShowModal(false);
    };
    const [activePage, setActivePage] = useState(() => {
        if (process.env.NODE_ENV === 'production') {
            // Código específico para el entorno de desarrollo
            return 1
        } else if (process.env.NODE_ENV === 'development') {
            // Código específico para el entorno de producción
            return 1
        }
    })
    const changeActivePage = (page) => {
        setActivePage(page)
    }
    return (
        <div id="homeDiv" className={classes.homeDiv}>
            {showModal && <FightModal showModal={showModal} onStartAdventure={handleStartAdventure} />}
            {activePage && activePage === 1 && !showModal && <Multiverse changeActivePage={changeActivePage}></Multiverse>}
            {activePage && activePage === 2 && !showModal && <Battleground changeActivePage={changeActivePage}></Battleground>}
        </div>
    )
}

export default Home