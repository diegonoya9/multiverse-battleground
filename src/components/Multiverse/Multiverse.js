import { memo, useState, useEffect, useContext } from "react"
import FightersPage from "./FightersPage.js"
import ObjectsPage from "./ObjectsPage.js"
import UsersPage from './UsersPage.js'
import classes from './Multiverse.module.css'
import ShopPage from "./ShopPage.js"
import ReactAudioPlayer from 'react-audio-player';
import musicFile from "../../assets/sounds/music/Aeroplane.WAV"
import Button from "../UI/Button.js"
import { MyContext } from '../../context/MyContext';

const Multiverse = ({ changeActivePage }) => {
    const audioStyle = {
        display: 'none', // Oculta el reproductor de audio visualmente
    };
    const { userContext, setUserId } = useContext(MyContext);
    let activeUser = userContext.idUsuario
    const [multiverseActivePage, setMultiverseActivePage] = useState("mainMenu")
    const changeMultiverseActivePage = (activePage) => {
        setMultiverseActivePage(activePage)
    }
    const [user, setUser] = useState()
    const [money, setMoney] = useState()
    useEffect(() => {
        if (user) {
            let newMoney = user.objects.filter((object) => {
                return object.name === "money"
            })
            setMoney(newMoney[0].quantity)
        }
    }, [user])
    const updateUser = () => {
        fetch('https://multiverse-battleground-default-rtdb.firebaseio.com/users/' + activeUser + '.json')
            .then(response => response.json())
            .then(data => { setUser(data) })
    }
    useEffect(() => {
        fetch('https://multiverse-battleground-default-rtdb.firebaseio.com/users/' + activeUser + '.json')
            .then(response => response.json())
            .then(data => { setUser(data) })
        const audio = document.getElementById('audioPlayer');
        if (audio) {
            audio.play()
        }
    }, [])
    useEffect(() => {
        fetch('https://multiverse-battleground-default-rtdb.firebaseio.com/users/' + activeUser + '.json')
            .then(response => response.json())
            .then(data => { setUser(data) })
    }, [multiverseActivePage])
    return (
        <div className={classes.container}>
            {multiverseActivePage === 'mainMenu' && user && (
                <div className={`${classes.mainMenu} ${classes.container}`}>
                    <ReactAudioPlayer src={musicFile} autoPlay id="audioPlayer" controls style={audioStyle} />
                    <h1 className={classes.h1}>Welcome {user.name}</h1>
                    <h2 className={classes.h2}>You have {money} pesos</h2>
                    <Button value='Fight' colorType="red" onClick={() => changeActivePage(2)}></Button>
                    <Button value='Fighters' colorType="brown" onClick={(e) => {
                        changeMultiverseActivePage('fighters');
                    }}></Button>
                    <Button value='Bag' colorType="green" onClick={(e) => {
                        changeMultiverseActivePage('bag');
                    }}></Button>
                    <Button value='Shop' colorType="yellow" onClick={(e) => {
                        changeMultiverseActivePage('shop');
                    }}></Button>
                    <Button value='Users' colorType="blue" onClick={(e) => {
                        changeMultiverseActivePage('users');
                    }}></Button>
                </div>
            )}
            {multiverseActivePage === "fighters" && <FightersPage
                updateUser={updateUser} changeMultiverseActivePage={changeMultiverseActivePage} user={user}></FightersPage>}
            {multiverseActivePage === "bag" && <ObjectsPage changeMultiverseActivePage={changeMultiverseActivePage} user={user}></ObjectsPage>}
            {multiverseActivePage === "shop" && <ShopPage changeMultiverseActivePage={changeMultiverseActivePage} ></ShopPage>}
            {multiverseActivePage === "users" && <UsersPage changeMultiverseActivePage={changeMultiverseActivePage} ></UsersPage>}
        </div>

        /*<div>
            {multiverseActivePage === "mainMenu" && user &&
                <div>
                    <ReactAudioPlayer src={musicFile} autoPlay id="audioPlayer" controls style={audioStyle} />
                    <h1 className={classes.h1}>Welcome {user.name}</h1>
                    <h2 className={classes.h2}>You have {money} pesos</h2>
                    <input type="submit" onClick={(e) => { changeActivePage(2); }} value={'Fight'} />
                    <input type="submit" onClick={(e) => { changeMultiverseActivePage("fighters") }} value={'Fighters'} />
                    <input type="submit" onClick={(e) => { changeMultiverseActivePage("bag") }} value={'Bag'} />
                    <input type="submit" onClick={(e) => { changeMultiverseActivePage("shop") }} value={'Shop'} />
                </div>
            }
            {multiverseActivePage === "fighters" && <FightersPage changeMultiverseActivePage={changeMultiverseActivePage} user={user}></FightersPage>}
            {multiverseActivePage === "bag" && <ObjectsPage changeMultiverseActivePage={changeMultiverseActivePage} user={user}></ObjectsPage>}
            {multiverseActivePage === "shop" && <ShopPage changeMultiverseActivePage={changeMultiverseActivePage} ></ShopPage>}
        </div>*/
    )
}
export default memo(Multiverse)