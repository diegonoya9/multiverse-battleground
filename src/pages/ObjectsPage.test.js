// ObjectsPage.test.js
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { MyContextProvider } from '../context/MyContext';
import ObjectsPage from './ObjectsPage';
import renderer from 'react-test-renderer'
import { createBrowserRouter, RouterProvider } from 'react-router-dom'
import RootLayuout from './Root';
import Home from '../components/Multiverse/Home';
import Multiverse from '../components/Multiverse/Multiverse';
import { initReactI18next, I18nextProvider } from 'react-i18next';
import i18n from '../i18n';
jest.mock('react-audio-player', () => {
  const ReactAudioPlayer = jest.fn();
  ReactAudioPlayer.prototype.play = jest.fn();
  return ReactAudioPlayer;
});
/*const objects = [
  {
    "user_object_id": 2,
    "user_id": 1,
    "object_id": 7,
    "quantity": 25911,
    "objects": {
      "object_id": 7,
      "category": "userItem",
      "description": "Money",
      "img": "./assets/img/money.png",
      "name": "Money",
      "price": 1,
      "quantity": 1,
      "type": "consumable",
      "actionobjects": []
    },
    "name": "Money",
    "description": "Money",
    "img": "./assets/img/money.png"
  }
]*/
const mockNavigate = jest.fn();
beforeAll(() => {
  i18n.use(initReactI18next)
});
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('ObjectsPage Component', () => {
  // Mock del contexto
  const mockContext = {
    userContext: {
      backEndUrl: 'http://example.com',
      bg: 0,
      user: {
        user_id: 7
      }
    },
    setUserId: jest.fn(),
  };

  it('handles click on "Back to Main Menu" button', async () => {
    render(
      <MyContextProvider value={mockContext}>
        <ObjectsPage />
      </MyContextProvider>
    );

    // Simula el clic en el botón
    fireEvent.click(screen.getByText('Back to Main Menu'));

    // Asegúrate de que la función navigate se haya llamado con "/"
    expect(mockNavigate).toHaveBeenCalledWith('/');
    // Puedes realizar más assertions según tus necesidades
  });
  it('renders objectpage correctly according to snapshot', () => {
    const tree = renderer
      .create(
        <MyContextProvider value={mockContext}>
          <RouterProvider router={testRouter}>
            <ObjectsPage />
          </RouterProvider>
        </MyContextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Puedes escribir más pruebas para otras interacciones y comportamientos del componente
});

/*
afterEach(() => {
  global.fetch.mockRestore();
  cleanup();
});*/
const testRouter = createBrowserRouter([
  {
    path: '/',
    element: <RootLayuout />,
    children: [
      {
        path: '/',
        element: <Home></Home>,
        children: [
          { index: true, element: <Multiverse /> },
          { path: '/bag', element: <ObjectsPage /> },
        ]
      }
    ]
  }
])

/*
let component
const user = { user_id: 1 }
test('renders objects page component', async () => {
  global.fetch = jest.fn().mockResolvedValue({
    json: jest.fn().mockResolvedValue(objects)
  });
  // Renderiza ObjectsPage dentro de MyContextProvider con el contexto simulado
  await act(async () => {
    component = render(
      <RouterProvider router={testRouter}>
        <MyContextProvider value={{ userContext: mockUserContext }}>
          <ObjectsPage />
        </MyContextProvider>
      </RouterProvider>
    );
  })

  // Puedes agregar expectativas para asegurarte de que los elementos esperados est�n presentes
  component.getByText("Back to Main Menu")
});
it('renders objectpage correctly according to snapshot', () => {
  const tree = renderer
    .create(
      <MyContextProvider value={{ userContext: mockUserContext }}>
        <RouterProvider router={testRouter}>
          <ObjectsPage />
        </RouterProvider>
      </MyContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
*/