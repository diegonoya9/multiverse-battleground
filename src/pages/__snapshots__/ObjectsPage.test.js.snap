// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectsPage Component renders objectpage correctly according to snapshot 1`] = `
Array [
  <div>
    <link
      href="https://fonts.googleapis.com"
      rel="preconnect"
    />
    <link
      href="https://fonts.gstatic.com"
      rel="preconnect"
    />
    <link
      href="https://fonts.googleapis.com/css2?family=Allerta+Stencil&family=Merienda&family=Mochiy+Pop+One&family=Dhurjati&family=Black+Ops+One&family=Orbitron&display=swap"
      rel="stylesheet"
    />
  </div>,
  "0",
  <div
    className="homeDivGoogle"
    id="homeDivGoogle"
  >
    <div
      className="divh1Login"
    >
      <h1
        className="h1Login"
      >
        Login with Google
      </h1>
    </div>
    <div
      style={
        Object {
          "height": 40,
        }
      }
    />
     
  </div>,
]
`;
